src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,57 @@
+import { Link, useSearchParams } from 'react-router-dom';
+
+import classNames from 'classnames';
+import React from 'react';
+import { getSearchWith } from '../utils/searchHelper';
+import { SexOptions } from '../types/SexParams';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || null;
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchParams(
+      getSearchWith(searchParams, { query: event.target.value || null }),
+    );
+  };
+
+  const toggleCentury = (century: string) => {
+    return centuries.includes(century)
+      ? centuries.filter(filterCenturies => century !== filterCenturies)
+      : [...centuries, century];
+  };
```

Comment:
The `toggleCentury` function is designed to add or remove a century from the `centuries` array. However, it returns a new array but does not update the `searchParams` with this new array. Ensure that the updated centuries array is set back to the search parameters to reflect changes in the URL.